lang/cs/Org.Apache.REEF.Tang.Tests/Injection/TestInjection.cs

```diff
@@ -304,6 +304,55 @@ public void TestExternalObject()
 
             Assert.IsNotNull(o.ExternalObject is ExternalClass);
         }
+        
+        /// <summary>
+        /// In this test, interface is generic of T. Implementations use different T as type. 
+        /// When doing injection, we need to get type of generic type T so that to get generic interface
+        /// </summary>
+        [TestMethod]
+        public void TestInjectionWithT()
```

Comment:
I think I now understand what is being tested here, but it isn't really clear from the code. Here is my attempt at a description:

> Given a generic Interface `A<T>` and and implementation of it `B<T>`, this test validates that instances of `B` can be created when `T` is in fact constructed at runtime from a `string`. More specifically, `Injector.GetInstance(A<T>)` shall return a correctly typed instance of `B` when `T` is known only at runtime.

If this is indeed the fact tested here, I believe it can be done in less code. For that test, it is immaterial whether the `string` representation of `T` was part of a Configuration or not. So the lines dealing with `MessageType` can be removed and replaced with `string` instances. Also, the instantiation of instances of `A` with statically known type parameter `T` are redundant with other tests and can be removed.

What I am not sure of is what the desired behavior is when there is no implementation of `A<T>` for a specific `T` in the configuration. Is this supposed to result in an error? Or can a user bind an implementation for `A` for any `T`, and Tang would instantiate that implementation for any given value of `T`? It would be good to also have a test for that as clarification.